main/test/flix/Test.Exp.MatchType.flix

```diff
@@ -0,0 +1,287 @@
+namespace Test/Exp/ReifyType {
```

Comment:
These test cases don't seem to use much polymorphism. I think we want to call functions that take type parameters and reflect on those, right?

Reply:
I think at minimum we should send the values through the identity function.

Reply:
We want to see that the Monomorph correctly clones functions and eliminates cases with a healthy mix of different types.